pkg/apis/core/v1alpha1/federatedtypeconfig_types.go

```diff
@@ -31,31 +31,27 @@ type FederatedTypeConfigSpec struct {
 	// groupName fields will be set from the metadata.name of this resource. The
 	// kind field must be set.
 	Target APIResource `json:"target"`
-	// Whether or not the target type is namespaced. The federation types
-	// (template, placement, overrides) for the type will share this
+	// Whether or not the target type is namespaced. The federation
+	// types (FederatedType, Status) for the type will share this
 	// characteristic.
+	//
+	// TODO(marun) Remove in favor of using the value from Target and
+	// FederatedType (depending on context).
 	Namespaced bool `json:"namespaced"`
 	// Which field of the target type determines whether federation
 	// considers two resources to be equal.
+	//
+	// TODO(marun) Remove and discover the field to use at runtime.
 	ComparisonField common.VersionComparisonField `json:"comparisonField"`
 	// Whether or not propagation to member clusters should be enabled.
 	PropagationEnabled bool `json:"propagationEnabled"`
-	// Configuration for the template type that holds the base definition of
-	// a federated resource.
-	Template APIResource `json:"template"`
-	// Configuration for the placement type that holds information about which
-	// member clusters the resource should be federated to. If not provided, the
-	// group and version will default to those provided for the template
-	// resource.
-	Placement APIResource `json:"placement"`
-	// Configuration for the override type that holds information about how the
-	// resource should be changed from the template when in certain member
-	// clusters. If not provided, the group and version will default to those
-	// provided for the template resource.
-	Override APIResource `json:"override"`
+	// Configuration for the federated type that defines how the target
+	// type should be propagated to multiple clusters.
+	FederatedType APIResource `json:"federatedType"`
```

Comment:
Is `Federated` prefix necessary here? This is already inside `FederatedTypeConfigSpec`.
My suggestion would be to try and reduce our usage of `Federated` prefix where ever logically possible. The code base already has an overdose of the same :).

Reply:
I agree that 'federated' is an overused prefix, but I don't think `Type` is sufficiently descriptive.  I'm happy to consider an alternate name, `FederatedType` is just a sane default.

Reply:
Lets continue with `FederatedType`. Thanks.